rfcs/text/0032-structured-benchmark-programs.md

```diff
@@ -0,0 +1,157 @@
+Similarly, programs that require classical input data such as rotation angles use input parameters to allow users to specify the data at runtime.
+An example of this might be a VQA ansatz that is repeated several times using `for` loops.
+
+In contrast, some programs are provided with fixed sizes and parameters, either because they cannot easily be generalized or because they are intended as simpler test scenarios for basic compiler functionality.
+Examples of this include simple Quantum Teleportation programs, or predefined instances of Grover's search algorithm.
+
+The modular directory structure allows future adaptations and extensions of the benchmark suite, either by adding other structured programs or by also including entirely new categories of benchmarks.
+
+While this feature may not directly impact the way many users interact with Jeff, it provides a valuable resource for evaluating and improving quantum compilers and test their compatibility with Jeff and advanced quantum programming features.
+
+# Implementation-level explanation
+[implementation-level-explanation]: #implementation-level-explanation
+
+The following list outlines the proposed structured benchmark programs to be included in the `benchmarks/structured` directory, along with brief description, grouped by the way they utilize structured control flow:
+
+*Note: this list was created from previous discussions among community members and is open for further suggestions and modifications.*
+
+## Static Loops without Dynamic Qubit Indexing
+
+*This class of benchmark programs utilizes loops (such as `for` and `while`) that **do not** depend on dynamic execution results. Inside the loop bodies, all operations are still statically indexed.*
+
+- *Grover's Search Algorithm*: Oracle and diffusion operations can be applied repeatedly using a fixed-size loop. (https://arxiv.org/abs/quant-ph/9605043)
+- *VQE Ansatz with Fixed Repetitions*: A variational ansatz circuit that applies a set of parameterized gates in a loop with a predetermined number of repetitions. ([Peruzzo et al., 2014](https://arxiv.org/abs/1304.3061))
+- *QAOA with Fixed Repetitions*: A Quantum Approximate Optimization Algorithm circuit that applies problem and mixer Hamiltonians in a loop with a fixed number of layers. ([Farhi et al., 2014](https://arxiv.org/abs/1411.4028))
+
+## Static Loops with Dynamic Qubit Indexing
+
+*This class of benchmark programs utilizes loops (such as `for` and `while`) that **do not** depend on dynamic execution results. Inside the loop bodies, operations may be dynamically indexed (e.g. based on the loop parameter).*
+
+- *GHZ State Preparation*: A static loop is used to entangle all qubits with each other using `cx` gates where, at each loop index `i`, the `cx` gate is applied between qubit `0` and qubit `i`. ([Wikipedia on GHZ states](https://en.wikipedia.org/wiki/GHZ_state))
```

Comment:
Just noting this here: An alternative construction for the GHZ state would be to entangle `i` with `i+1` for `i` from 0 to n-2.
Gives a linear topology circuit compared to a star topology circuit.
Might be worth to include both?